### PR TITLE
Practice2

### PR DESCRIPTION
**Pull request by ne1true**

### CODE REVIEW

File: CatsList/CatsListTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        viewModel.calculateNumberOfSections()
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        viewModel.calculateNumberOfRows(section)
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.section == 0 {
```
Comment (vasilevkin) [inline CatsList/CatsListTableViewController.swift:39]: better use switch

File: CatsList/NetworkManager/NetworkManager.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    func loadCatsList(from url: String, complitionHandler: @escaping ([Cats]) -> ())
+    func loadCatsImageData(url: String?, complitionHandler: @escaping (Data?) -> ())
+    
+}
+
+class NetworkManagerImpl: NetworkManager {
+    
+    func loadCatsList(from url: String, complitionHandler: @escaping ([Cats]) -> ()) {
+        let session = URLSession.shared
+        session.configuration.timeoutIntervalForRequest = 30
```
Comment (vasilevkin) [inline CatsList/NetworkManager/NetworkManager.swift:21]: move magic numbers into constants

File: CatsList/NetworkManager/NetworkManager.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        guard let url = URL(string: url) else {
+            return
+        }
+        let task = session.dataTask(with: url) { data, response, error in
+            if error != nil {
+                if (error as? URLError)?.code == .timedOut {
+                    print("Timeout error!")
```
Comment (vasilevkin) [inline CatsList/NetworkManager/NetworkManager.swift:28]: show these errors to user

**Review by vasilevkin — CHANGES_REQUESTED** [review]
good start.
continue with implementation.

File: CatsList/CatsListTableViewController.swift (hunk 1 above)
Comment (ne1true) [reply CatsList/CatsListTableViewController.swift]: done in pr4

File: CatsList/NetworkManager/NetworkManager.swift (hunk 3 above)
Comment (ne1true) [reply CatsList/NetworkManager/NetworkManager.swift]: done in pr4

Comment (ne1true) [reply CatsList/NetworkManager/NetworkManager.swift]: done in pr4

File: CatsList/NetworkManager/NetworkManager.swift (hunk 2 above)
Comment (ne1true) [reply CatsList/NetworkManager/NetworkManager.swift]: done in pr4

**Pull request merged**